src/components/Timetable.jsx

```diff
@@ -31,190 +39,151 @@ const propTypes = {
   screenSize: PropTypes.number,
 }
 
-class Timetable extends React.Component {
+function calculateEvents (events, timeline) {
+  return R.map((event) => {
+    const {position, length} = calculateEventPosition(event, timeline)
+    return {
+      ...event,
+      _position: position,
+      _length: length,
+      _appear: eventAppearance(event),
+    }
+  }, events)
+}
 
-  /**
-   * Hides the days element by removing its animation property class
-   */
-  hide () {
-    const el = this.refs.days
+function createHourLabels (timeline, layout) {
+  console.log(timeline)
+  return calculateHourLabels(timeline).map((label) => {
+    console.log(label)
+    return <HourLabel key={label.id} position={label.position} length={label.length} layout={layout}>
+      {label.label}
+    </HourLabel>
+  })
+}
 
-    // Replay CSS animation
-    el.classList.remove('a-left')
-    el.classList.remove('a-right')
-  }
+function createDays (props, dayCount, events, eventsfn, animationDirection) {
```

Comment:
- Expand `props` if possible.
- Why are you passing in `eventsfn` instead of calling `createDayEvents` directly?